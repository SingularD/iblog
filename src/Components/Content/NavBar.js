import React from 'react'
import { Link } from "react-router-dom";
import { navbarData } from "../../Data";

import GithubUrl from '../../assets/images/github.png'
import RssUrl from '../../assets/images/rss.png'
import SearchUrl from '../../assets/images/search.png'


class NavBar extends React.Component{
  constructor(props) {
    super(props)
    this.state = {
      portraitUrl: navbarData.portraitUrl,
      username: navbarData.username,
      resume: navbarData.resume,
      githubUrl: navbarData.githubUrl,
      rssUrl: navbarData.rssUrl
    }
  }

  render() {
    return (
        <div
          className="col-lg-3 bg-light
          d-lg-block nav-bar"
        >
          <div className="nav-bar-container">
            <div className="nav-bar-head-portrait">
              <img src={this.state.portraitUrl} alt="头像"/>
            </div>
            <div className="nav-bar-resume">
              <h3>{this.state.username}</h3>
              <p>{this.state.resume}</p>
            </div>
            <div className="nav-bar-links">
              <ul className="list-group list-group-flush">
                <li className="list-group-item border-0"><Link to="/">首页</Link></li>
                <li className="list-group-item border-0"><Link to="/archive">归档</Link></li>
                <li className="list-group-item border-0"><Link to="/category">专题</Link></li>
                <li className="list-group-item border-0"><Link to="/friend-link">友链</Link></li>
                <li className="list-group-item border-0"><Link to="/about">关于</Link></li>
              </ul>
            </div>
            <div className="nav-bar-hrefs">
              <a href={this.state.githubUrl}>
                <img src={GithubUrl} alt="github"/>
              </a>
              <img src={RssUrl} alt="rss"/>
              <Link to="/search">
                <img src={SearchUrl} alt="search"/>
              </Link>
            </div>
          </div>
        </div>
    )
  }
}

export default NavBar
