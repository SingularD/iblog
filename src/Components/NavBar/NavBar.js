import React from 'react'
import PropTypes from 'prop-types'
import { Link } from "react-router-dom";

import PortraitUrl from '../../assets/images/portrait_mode.png'
import GithubUrl from '../../assets/images/github.png'
import RssUrl from '../../assets/images/rss.png'
import SearchUrl from '../../assets/images/search.png'


class NavBar extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
        <div
          className="col-lg-3 bg-light
          d-lg-block nav-bar"
        >
          <div className="nav-bar-container">
            <div className="nav-bar-head-portrait">
              <img src={PortraitUrl} alt="头像"/>
            </div>
            <div className="nav-bar-resume">
              <h3>姓名</h3>
              <p>这是一段个人简介</p>
            </div>
            <div className="nav-bar-links">
              <ul className="list-group list-group-flush">
                <li className="list-group-item border-0"><Link to="/">首页</Link></li>
                <li className="list-group-item border-0"><Link to="/archive">归档</Link></li>
                <li className="list-group-item border-0"><Link to="/category">专题</Link></li>
                <li className="list-group-item border-0"><Link to="/friend-link">友链</Link></li>
                <li className="list-group-item border-0"><Link to="/about">关于</Link></li>
              </ul>
            </div>
            <div className="nav-bar-hrefs">
              <img src={GithubUrl} alt="github"/>
              <img src={RssUrl} alt="rss"/>
              <Link to="search">
                <img src={SearchUrl} alt="search"/>
              </Link>
            </div>
          </div>
        </div>
    )
  }
}

export default NavBar
