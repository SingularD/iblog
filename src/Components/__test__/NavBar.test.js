import React from 'react'
import NavBar from '../Content/NavBar'
import { shallow } from "enzyme";

import PortraitUrl from '../../assets/images/portrait_mode.png'
import GithubUrl from '../../assets/images/github.png'
import RssUrl from '../../assets/images/rss.png'
import SearchUrl from '../../assets/images/search.png'

describe("测试导航栏是否正常渲染", () => {
  let wrapper = shallow(<NavBar />)
  it("导航栏快照匹配", () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("检测导航头像以及相关链接图片是否正常渲染", () => {
    expect(wrapper.find('.nav-bar-head-portrait img').props().src)
      .toEqual(PortraitUrl)
    expect(wrapper.find('.nav-bar-hrefs img').length).toEqual(3)
    expect(wrapper.find('.nav-bar-hrefs img').at(0).props().src)
      .toEqual(GithubUrl)
    expect(wrapper.find('.nav-bar-hrefs img').at(1).props().src)
      .toEqual(RssUrl)
    expect(wrapper.find('.nav-bar-hrefs img').at(2).props().src)
      .toEqual(SearchUrl)
  })
  it("检测导航栏文字部分是否正常渲染", () => {
    expect(wrapper.find('.nav-bar-resume h3').text()).toEqual('Singular')
    expect(wrapper.find('.nav-bar-resume p').text()).toEqual('这是一段个人简介')
    expect(wrapper.find('.list-group-item').length).toEqual(5)
  })
})
