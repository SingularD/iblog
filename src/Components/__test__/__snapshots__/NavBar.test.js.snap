// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试导航栏是否正常渲染 导航栏快照匹配 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-bar-container"
      >
        <div
          className="nav-bar-head-portrait"
        >
          <img
            alt="头像"
            src="portrait_mode.png"
          />
        </div>
        <div
          className="nav-bar-resume"
        >
          <h3>
            Singular
          </h3>
          <p>
            这是一段个人简介
          </p>
        </div>
        <div
          className="nav-bar-links"
        >
          <ul
            className="list-group list-group-flush"
          >
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/"
              >
                首页
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/archive"
              >
                归档
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/category"
              >
                专题
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/friend-link"
              >
                友链
              </Link>
            </li>
            <li
              className="list-group-item border-0"
            >
              <Link
                to="/about"
              >
                关于
              </Link>
            </li>
          </ul>
        </div>
        <div
          className="nav-bar-hrefs"
        >
          <img
            alt="github"
            src="github.png"
          />
          <img
            alt="rss"
            src="rss.png"
          />
          <Link
            to="/search"
          >
            <img
              alt="search"
              src="search.png"
            />
          </Link>
        </div>
      </div>,
      "className": "col-lg-3 bg-light d-lg-block nav-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nav-bar-head-portrait"
          >
            <img
              alt="头像"
              src="portrait_mode.png"
            />
          </div>,
          <div
            className="nav-bar-resume"
          >
            <h3>
              Singular
            </h3>
            <p>
              这是一段个人简介
            </p>
          </div>,
          <div
            className="nav-bar-links"
          >
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/"
                >
                  首页
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/archive"
                >
                  归档
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/category"
                >
                  专题
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/friend-link"
                >
                  友链
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/about"
                >
                  关于
                </Link>
              </li>
            </ul>
          </div>,
          <div
            className="nav-bar-hrefs"
          >
            <img
              alt="github"
              src="github.png"
            />
            <img
              alt="rss"
              src="rss.png"
            />
            <Link
              to="/search"
            >
              <img
                alt="search"
                src="search.png"
              />
            </Link>
          </div>,
        ],
        "className": "nav-bar-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="头像"
              src="portrait_mode.png"
            />,
            "className": "nav-bar-head-portrait",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "头像",
              "src": "portrait_mode.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Singular
              </h3>,
              <p>
                这是一段个人简介
              </p>,
            ],
            "className": "nav-bar-resume",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Singular",
              },
              "ref": null,
              "rendered": "Singular",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "这是一段个人简介",
              },
              "ref": null,
              "rendered": "这是一段个人简介",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/"
                >
                  首页
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/archive"
                >
                  归档
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/category"
                >
                  专题
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/friend-link"
                >
                  友链
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/about"
                >
                  关于
                </Link>
              </li>
            </ul>,
            "className": "nav-bar-links",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/"
                  >
                    首页
                  </Link>
                </li>,
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/archive"
                  >
                    归档
                  </Link>
                </li>,
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/category"
                  >
                    专题
                  </Link>
                </li>,
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/friend-link"
                  >
                    友链
                  </Link>
                </li>,
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/about"
                  >
                    关于
                  </Link>
                </li>,
              ],
              "className": "list-group list-group-flush",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/"
                  >
                    首页
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "首页",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "首页",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/archive"
                  >
                    归档
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "归档",
                    "to": "/archive",
                  },
                  "ref": null,
                  "rendered": "归档",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/category"
                  >
                    专题
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "专题",
                    "to": "/category",
                  },
                  "ref": null,
                  "rendered": "专题",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/friend-link"
                  >
                    友链
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "友链",
                    "to": "/friend-link",
                  },
                  "ref": null,
                  "rendered": "友链",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    to="/about"
                  >
                    关于
                  </Link>,
                  "className": "list-group-item border-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "关于",
                    "to": "/about",
                  },
                  "ref": null,
                  "rendered": "关于",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="github"
                src="github.png"
              />,
              <img
                alt="rss"
                src="rss.png"
              />,
              <Link
                to="/search"
              >
                <img
                  alt="search"
                  src="search.png"
                />
              </Link>,
            ],
            "className": "nav-bar-hrefs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "github",
                "src": "github.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "rss",
                "src": "rss.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="search"
                  src="search.png"
                />,
                "to": "/search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "search",
                  "src": "search.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-bar-container"
        >
          <div
            className="nav-bar-head-portrait"
          >
            <img
              alt="头像"
              src="portrait_mode.png"
            />
          </div>
          <div
            className="nav-bar-resume"
          >
            <h3>
              Singular
            </h3>
            <p>
              这是一段个人简介
            </p>
          </div>
          <div
            className="nav-bar-links"
          >
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/"
                >
                  首页
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/archive"
                >
                  归档
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/category"
                >
                  专题
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/friend-link"
                >
                  友链
                </Link>
              </li>
              <li
                className="list-group-item border-0"
              >
                <Link
                  to="/about"
                >
                  关于
                </Link>
              </li>
            </ul>
          </div>
          <div
            className="nav-bar-hrefs"
          >
            <img
              alt="github"
              src="github.png"
            />
            <img
              alt="rss"
              src="rss.png"
            />
            <Link
              to="/search"
            >
              <img
                alt="search"
                src="search.png"
              />
            </Link>
          </div>
        </div>,
        "className": "col-lg-3 bg-light d-lg-block nav-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="nav-bar-head-portrait"
            >
              <img
                alt="头像"
                src="portrait_mode.png"
              />
            </div>,
            <div
              className="nav-bar-resume"
            >
              <h3>
                Singular
              </h3>
              <p>
                这是一段个人简介
              </p>
            </div>,
            <div
              className="nav-bar-links"
            >
              <ul
                className="list-group list-group-flush"
              >
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/"
                  >
                    首页
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/archive"
                  >
                    归档
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/category"
                  >
                    专题
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/friend-link"
                  >
                    友链
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/about"
                  >
                    关于
                  </Link>
                </li>
              </ul>
            </div>,
            <div
              className="nav-bar-hrefs"
            >
              <img
                alt="github"
                src="github.png"
              />
              <img
                alt="rss"
                src="rss.png"
              />
              <Link
                to="/search"
              >
                <img
                  alt="search"
                  src="search.png"
                />
              </Link>
            </div>,
          ],
          "className": "nav-bar-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="头像"
                src="portrait_mode.png"
              />,
              "className": "nav-bar-head-portrait",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "头像",
                "src": "portrait_mode.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Singular
                </h3>,
                <p>
                  这是一段个人简介
                </p>,
              ],
              "className": "nav-bar-resume",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Singular",
                },
                "ref": null,
                "rendered": "Singular",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "这是一段个人简介",
                },
                "ref": null,
                "rendered": "这是一段个人简介",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="list-group list-group-flush"
              >
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/"
                  >
                    首页
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/archive"
                  >
                    归档
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/category"
                  >
                    专题
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/friend-link"
                  >
                    友链
                  </Link>
                </li>
                <li
                  className="list-group-item border-0"
                >
                  <Link
                    to="/about"
                  >
                    关于
                  </Link>
                </li>
              </ul>,
              "className": "nav-bar-links",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="list-group-item border-0"
                  >
                    <Link
                      to="/"
                    >
                      首页
                    </Link>
                  </li>,
                  <li
                    className="list-group-item border-0"
                  >
                    <Link
                      to="/archive"
                    >
                      归档
                    </Link>
                  </li>,
                  <li
                    className="list-group-item border-0"
                  >
                    <Link
                      to="/category"
                    >
                      专题
                    </Link>
                  </li>,
                  <li
                    className="list-group-item border-0"
                  >
                    <Link
                      to="/friend-link"
                    >
                      友链
                    </Link>
                  </li>,
                  <li
                    className="list-group-item border-0"
                  >
                    <Link
                      to="/about"
                    >
                      关于
                    </Link>
                  </li>,
                ],
                "className": "list-group list-group-flush",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      to="/"
                    >
                      首页
                    </Link>,
                    "className": "list-group-item border-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "首页",
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "首页",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      to="/archive"
                    >
                      归档
                    </Link>,
                    "className": "list-group-item border-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "归档",
                      "to": "/archive",
                    },
                    "ref": null,
                    "rendered": "归档",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      to="/category"
                    >
                      专题
                    </Link>,
                    "className": "list-group-item border-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "专题",
                      "to": "/category",
                    },
                    "ref": null,
                    "rendered": "专题",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      to="/friend-link"
                    >
                      友链
                    </Link>,
                    "className": "list-group-item border-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "友链",
                      "to": "/friend-link",
                    },
                    "ref": null,
                    "rendered": "友链",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      to="/about"
                    >
                      关于
                    </Link>,
                    "className": "list-group-item border-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "关于",
                      "to": "/about",
                    },
                    "ref": null,
                    "rendered": "关于",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="github"
                  src="github.png"
                />,
                <img
                  alt="rss"
                  src="rss.png"
                />,
                <Link
                  to="/search"
                >
                  <img
                    alt="search"
                    src="search.png"
                  />
                </Link>,
              ],
              "className": "nav-bar-hrefs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "github",
                  "src": "github.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "rss",
                  "src": "rss.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <img
                    alt="search"
                    src="search.png"
                  />,
                  "to": "/search",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "search",
                    "src": "search.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
